src/main/java/edu/harvard/iq/dataverse/DatasetServiceBean.java

```diff
@@ -1042,4 +1061,492 @@ public void deleteHarvestedDataset(Dataset dataset, DataverseRequest request, Lo
+                            countError++;
+                        }
+                    }
+
+                    String newjsonData = jsonDataSecondAPI.build().toString();
+
+                    globusLogger.info("Successfully generated new JsonData for Second API call");
+
+
+                    String command = "curl -H \"X-Dataverse-key:" + token.getTokenString() + "\" -X POST " + httpRequestUrl + "/api/datasets/:persistentId/addFiles?persistentId=doi:" + datasetIdentifier + " -F jsonData='" + newjsonData + "'";
+                    System.out.println("*******====command ==== " + command);
+
+                    String output = addFilesAsync(command, globusLogger);
+                    if (output.equalsIgnoreCase("ok")) {
+                        //if(!taskSkippedFiles)
+                        if (countError == 0 ){
+                            userNotificationService.sendNotification((AuthenticatedUser) authUser, new Timestamp(new Date().getTime()), UserNotification.Type.GLOBUSUPLOADCOMPLETED, dataset.getId(), countSuccess + " files added out of "+ countAll , true);
+                        }
+                        else {
+                            userNotificationService.sendNotification((AuthenticatedUser) authUser, new Timestamp(new Date().getTime()), UserNotification.Type.GLOBUSUPLOADCOMPLETEDWITHERRORS, dataset.getId(), countSuccess + " files added out of "+ countAll , true);
+                        }
+                        globusLogger.info("Successfully completed api/datasets/:persistentId/addFiles call ");
+                    } else {
+                        globusLogger.log(Level.SEVERE, "******* Error while executing api/datasets/:persistentId/add call ", command);
+                    }
+
+                }
+
+                globusLogger.info("Files processed: " + countAll.toString());
+                globusLogger.info("Files added successfully: " + countSuccess.toString());
+                globusLogger.info("Files failures: " + countError.toString());
+                globusLogger.info("Finished upload via Globus job.");
+
+                if (fileHandlerSuceeded) {
+                    fileHandler.close();
+                }
+
+            } catch (Exception e) {
+                logger.info("Exception from globusUpload call ");
+                e.printStackTrace();
+                globusLogger.info("Exception from globusUpload call " + e.getMessage());
+            }
+        }
+    }
+
+    public static JsonObjectBuilder stringToJsonObjectBuilder(String str) {
+        JsonReader jsonReader = Json.createReader(new StringReader(str));
+        JsonObject jo = jsonReader.readObject();
+        jsonReader.close();
+
+        JsonObjectBuilder job = Json.createObjectBuilder();
+
+        for (Map.Entry<String, JsonValue> entry : jo.entrySet()) {
+            job.add(entry.getKey(), entry.getValue());
+        }
+
+        return job;
+    }
+
+    Executor executor = Executors.newFixedThreadPool(10);
```

Comment:
FWIW: I've used this (in the Bag creation code I think) but my understanding is that it is better practice to just use the @Asynchronous annotation. It has been a while since I looked, but I think you can also designate a specific new pool of threads in that case (i.e. if you don't want this task to compete with others in Dataverse for threads). The main advantage that I saw is that the size of such a pool is then configurable via payara rather than being hardcoded.